core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java

```diff
@@ -435,6 +432,34 @@ public void watchConfigFiles(BuildProducer<HotDeploymentWatchedFileBuildItem> wa
         }
     }
 
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void unknownConfigFiles(
+            ApplicationArchivesBuildItem applicationArchives,
+            LaunchModeBuildItem launchModeBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            ConfigRecorder configRecorder) throws Exception {
+
+        PathCollection rootDirectories = applicationArchives.getRootArchive().getRootDirectories();
+        if (!rootDirectories.isSinglePath()) {
+            return;
+        }
+
+        Set<String> buildTimeFiles = new HashSet<>();
+        buildTimeFiles.addAll(ConfigDiagnostic.configFiles(rootDirectories.getSinglePath()));
+        buildTimeFiles.addAll(ConfigDiagnostic.configFilesFromLocations());
+
+        // Report always at build time since config folder and locations may differ from build to runtime
+        ConfigDiagnostic.unknownConfigFiles(buildTimeFiles);
+
+        // TODO - Remove reflective registration after https://github.com/quarkusio/quarkus/pull/35247
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(URI.class).constructors(true).build());
+        // No need to include the application files, because they don't change
+        if (!launchModeBuildItem.getLaunchMode().isDevOrTest()) {
+            configRecorder.unknownConfigFiles();
```

Comment:
So you want to log it once at build time and one at runtime, right?

Reply:
Yes, because this also looks into the `config` folder and `quarkus.config.locations`, which may be different from build to runtime, so we need to perform the checks twice.